src/Maestro/Maestro.Web/Api/v2019_01_16/Controllers/BuildTimeController.cs

```diff
@@ -55,54 +57,38 @@ public virtual async Task<IActionResult> GetBuildTimes([Required]int id, int day
                 return NotFound();
             }
 
-            var parameters = new List<KustoParameter> {
-                new KustoParameter("_Repository", KustoDataTypes.String,  defaultChannel.Repository.Split("/").Last()),
-                new KustoParameter("_SourceBranch", KustoDataTypes.String, defaultChannel.Branch),
-                new KustoParameter("_Days", KustoDataTypes.TimeSpan, $"{days}d")
-            };
+            Dictionary<string, KustoQuery> queries = Helpers.CreateBuildTimesQueries(defaultChannel.Repository, defaultChannel.Branch, days);
 
-            // We only care about builds that complete successfully or partially successfully 
-            // from the given repository. We summarize duration of the builds over the last 7 days.
-            // There are multiple different definitions that run in parallel, so we summarize
-            // on the definition id and ultimately choose the definition that took the longest.
-            string commonQueryText = @"| where Repository endswith _Repository
-                | where Result != 'failed' and Result != 'canceled' 
-                | where FinishTime > ago(_Days) 
-                | extend duration = FinishTime - StartTime 
-                | summarize average_duration = avg(duration) by DefinitionId
-                | summarize max(average_duration)";
+            var results = await Task.WhenAll<IDataReader>(_kustoClientProvider.ExecuteKustoQueryAsync(queries["internal"]), 
```

Comment:
>internal [](start = 111, length = 8)

Should this be a strongly typed object rather than a dictionary? So that there isn't a typo opportunity.  Then it could just be queries.Internal